src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -647,11 +648,10 @@ public void run() {
                         numRetries = 0;
                     }
                 } catch (IOException e) {
-                    if (shutdown) {
```

Comment:
Update. I used to consider that the `closeSocket(client); ` should also be executed when exception was thrown.